runelite-client/src/main/java/net/runelite/client/plugins/gelogger/GELoggerPlugin.java

```diff
@@ -0,0 +1,144 @@
+ */
+package net.runelite.client.plugins.gelogger;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import lombok.extern.java.Log;
+import net.runelite.api.*;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.GrandExchangeOfferChanged;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+
+import javax.inject.Inject;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+
+@Log
+@PluginDescriptor(
+	name = "GE Logger",
+	description = "Logs all your activities in GE",
+	tags = {"grand", "exchange", "log"}
+)
+public class GELoggerPlugin extends Plugin
+{
+	HashMap<Integer, GEHistoryItem> latest_offers = new HashMap<>(8);
+	@Inject
+	private Client client;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	private GELoggerConfig config;
+
+	@Provides
+	GELoggerConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(GELoggerConfig.class);
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (event.getGameState() != GameState.LOGGED_IN)
+		{
+			latest_offers.clear();
+		}
+	}
+
+	@Subscribe
+	public void onGrandExchangeOfferChanged(GrandExchangeOfferChanged offerEvent)
+	{
+		GrandExchangeOffer offer = offerEvent.getOffer();
+		ItemComposition offerItem = itemManager.getItemComposition(offer.getItemId());
+		boolean buying = offer.getState() == GrandExchangeOfferState.BOUGHT
+				|| offer.getState() == GrandExchangeOfferState.BUYING
+				|| offer.getState() == GrandExchangeOfferState.CANCELLED_BUY;
+		if (offer.getState() == GrandExchangeOfferState.EMPTY)
+		{
+			logMerch(latest_offers.get(offerEvent.getSlot()));
+			latest_offers.remove(offerEvent.getSlot());
+		}
+		else
+		{
+			GEHistoryItem history_item = new GEHistoryItem(buying,  offerItem.getName(), offer.getItemId(), offer.getSpent(), offer.getQuantitySold());
+			latest_offers.put(offerEvent.getSlot(), history_item);
+		}
+	}
+
+	public void logMerch(GEHistoryItem item)
+	{
+		if (item != null && item.amount > 0)
+		{
+			try
+			{
+				BufferedWriter logger = new BufferedWriter(new FileWriter(config.logFile(), true));
+				logger.write(item.toString());
+				logger.newLine();
+				logger.close();
+				log.info(item.toString());
+			}
+			catch (IOException e)
+			{
+				e.printStackTrace();
+			}
+		}
+	}
+
+	public class GEHistoryItem
+	{
+		public Boolean bought;
+		public String item;
+		public Integer itemid;
+		public Integer total_price;
+		public Integer amount;
+
+		public GEHistoryItem(boolean bought, String item, int itemid, int total_price, int amount)
+		{
+			this.bought = bought;
+			this.item = item;
+			this.itemid = itemid;
+			this.total_price = total_price;
+			this.amount = amount;
+		}
+
+		@Override public String toString()
```

Comment:
.csv as in "character-separated values". It makes no sense to me to call it .tsv just because I use tab, as most csv libraries allow to change the delimiter, e.g. https://docs.python.org/2/library/csv.html

The only use case that comes to my mind is if you want to import it into excel. Otherwise I agree with you, that JSON would be better.

Reply:
csv is comma separated.
tsv is tab separated. 
If you wanted to go into it you could argue for dvs or delimiter separated value. But I digress.

The main point was why are we using this data format over the likes of JSON. As for the excel point, Excel (and so I'd presume competitors) are able to import JSON files.

Reply:
> So, for the next person finding this - the answer is no - LibreOffice Calc does not support direct import of JSON.

from https://stackoverflow.com/questions/22534203/does-libreoffice-calc-support-json-file-importing-sorting

But we could offer both options.